javascript/javascript2/week2/scripts/quizItems.js

```diff
@@ -0,0 +1,277 @@
+  form.appendChild(inputQuestion);
+
+  let optDiv = document.createElement("div");
+  optDiv.className = "optContainer";
+
+  for (let i = 1; i < 5; i++) {
+    let optLabel = document.createElement("label");
+    optLabel.className = "optLabel";
+    optLabel.setAttribute("for", `input-${i}-${number}`);
+    optLabel.id = `int-${i}-${number}`;
+    optLabel.textContent = `Possible Answer-${i}`;
+    // optLabel.appendChild(document.createElement("br"))
+    optDiv.appendChild(optLabel);
+
+    let ChoiceInput = document.createElement("input");
+    ChoiceInput.type = "text";
+    ChoiceInput.id = `ipt-${i}-${number}`;
+    ChoiceInput.name = `option-${i}`;
+    // ChoiceInput.appendChild(document.createElement("br"))
+    optDiv.appendChild(ChoiceInput);
+  }
+  let options = generateQuizOptions();
+
+  // let optionDiv = generateQuizElement()
+  form.appendChild(optDiv);
+  form.appendChild(options);
+  let saveButton = document.createElement("button");
+  saveButton.id = `saveButton-${number}`;
+  saveButton.textContent = "Save";
+  saveButton.type = "button";
+  // form.appendChild(optionDiv)
+  form.appendChild(saveButton);
+  main.appendChild(form);
+  body.appendChild(main);
+  generateQuizOptions();
+}
+
+function generateQuizOptions() {
+  let optionsDiv = document.createElement("div");
+  optionsDiv.className = "optionsContainer";
+
+  let titleOptions = document.createElement("h3");
+  titleOptions.className = "titleOptions";
+  optionsDiv.appendChild(titleOptions);
+  titleOptions.appendChild(document.createElement("br"));
+  let ulOptions = document.createElement("div");
+  ulOptions.className = "ulOptions";
+  optionsDiv.appendChild(ulOptions);
+  ulOptions.appendChild(document.createElement("br"));
+  let num = amountReceived;
+  for (let i = 1; i < 5; i++) {
+    let eachOpton = document.createElement("div");
+    eachOpton.className = "eachOption";
+    let optionLabel = document.createElement("label");
+    optionLabel.className = "optionLabel";
+    optionLabel.setAttribute("for", `input-${i}-${num}`);
+    optionLabel.id = `input-${i}-${num}`;
+    optionLabel.textContent = `is ${i} correct?`;
+    // optionLabel.appendChild(document.createElement("br"))
+    eachOpton.appendChild(optionLabel);
+
+    let optionCheckBoxInput = document.createElement("input");
+    optionCheckBoxInput.type = "checkbox";
+    optionCheckBoxInput.id = `input-${i}-${num}`;
+    optionCheckBoxInput.name = `Choice-${i}`;
+
+    eachOpton.appendChild(optionCheckBoxInput);
+    eachOpton.appendChild(document.createElement("br"));
+    num--;
+    ulOptions.appendChild(eachOpton);
+  }
+  return optionsDiv;
+}
+function saveDataForNow(form) {
+  let formData = new FormData(form);
+  let quizQuestions = {
+    id: 0,
+    question: "",
+    options: [],
+    correctAnswer: [],
+  };
+
+  let formValues = Array.from(formData.values());
+  let formDataEntries = Array.from(formData.entries());
+  let correctAnswerExtracted = formDataEntries[5];
```

Comment:
This code causes an error in the console if the user did not select a correct answer.